File: VB/Core/CommonVB.Core/Extensions/ServicesExtension.vb
Imports System.Runtime.CompilerServices

Public Module ServicesExtension

    <Extension>
    Public Function TryGetService(Of TModel As Class)(serviceProvider As IServiceProvider) As TModel

        Try

            Return serviceProvider.GetService(GetType(TModel))

        Catch ex As ObjectDisposedException

            ' ignore as we do not care...

        End Try

        Return Nothing

    End Function

End Module

File: CSharp/Core/Common.Core/Extensions/ServicesExtension.cs
namespace Common.Core.Extensions;

public static class ServicesExtension
{
    public static TModel? TryGetService<TModel>(this IServiceProvider serviceProvider) where TModel : class
    {
        try
        {
            return (TModel?)serviceProvider.GetService(typeof(TModel));
        }
        catch (ObjectDisposedException)
        {
            // ignore as we do not care...
        }

        return default;
    }
}